lib/src/domain/repository/repository_card.dart

```diff
@@ -1,28 +1,119 @@
 import 'dart:convert';
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:http/http.dart';
+import '../../core/util/constants_api_calls.dart';
+import '../../core/util/constants_strings.dart';
+import '../../data/datasource/local/DAOs/card_database.dart';
 import '../../data/datasource/remote/card_api_provider.dart';
 import '../../data/model/card.dart';
 
 class RepositoryCard {
   final CardApiProvider cardsApi;
   final Client client;
+  final CardDatabase cardDatabase;
+  late List<HearthstoneCard> cardListApi = <HearthstoneCard>[];
 
   RepositoryCard({
     required this.client,
+    required this.cardDatabase,
     required this.cardsApi,
   });
 
-  late List<HearthstoneCard> cardListApi = <HearthstoneCard>[];
+  Future<List<dynamic>> fetchDocumentSnapshot({
+    required String mainCollectionDoc,
+    required String subCollection,
+  }) async {
+    final List<dynamic> documentList = [];
+    final QuerySnapshot documentSnapshot = await cardDatabase.readItems(
+      mainCollectionDoc: mainCollectionDoc,
+      subCollectionDoc: subCollection,
+    );
+    final List<QueryDocumentSnapshot> docCards = documentSnapshot.docs;
+
+    for (final doc in docCards) {
+      documentList.add(
+        doc.data(),
+      );
+    }
+    if (documentList.isEmpty) {
+      throw Exception(kEmptyDatabase);
+    }
+    return documentList;
+  }
+
+  void addCardBase(
+    String endpoint,
+    List<dynamic> cardsApiList,
+  ) {
+    switch (endpoint) {
+      case kRacesDemonEndPointApi:
+        cardDatabase.addItem(
+          cardList: cardsApiList,
+          mainCollectionDoc: kMainCollectionDocDemon,
+          subCollectionDoc: kSubCollectionDemon,
+        );
+        break;
+      case kRacesTotemEndPointApi:
+        cardDatabase.addItem(
+          cardList: cardsApiList,
+          mainCollectionDoc: kMainCollectionDocTotem,
+          subCollectionDoc: kSubCollectionTotem,
+        );
+        break;
+      case kRacesMurlocEndPointApi:
+        cardDatabase.addItem(
+          cardList: cardsApiList,
+          mainCollectionDoc: kMainCollectionDocMurloc,
+          subCollectionDoc: kSubCollectionMurloc,
+        );
+        break;
+    }
+  }
 
   Future<List<HearthstoneCard>> addCardList(String endpoint) async {
     cardListApi = [];
-    final String responseBody = await cardsApi.fetchCardList(
-      endpoint,
-      client,
-    );
-    final List<dynamic> cardsApiList = await jsonDecode(
-      responseBody,
-    ) as List<dynamic>;
+    List<dynamic> cardsApiList = [];
+    try {
+      final responseBody = await cardsApi.fetchCardList(
+        endpoint,
+        client,
+      );
+      cardsApiList = await jsonDecode(
```

Comment:
This await is necessary?

Reply:
No, the call has been already made and completed by that point. Done